### PR TITLE
[Feat] Refactor rejection sampler

### PR DESCRIPTION
**Pull request by realliujiaxu**
### What this PR does / why we need it?

Currently, we are using `AscendRejctionSampler` that extends from `RejctionSampler` in spec decoding. `AscendRejctionSampler` override `forward` of `RejctionSampler`, only aming to replace `rejection_sample` func. This   
 causes a lot of code of `RejctionSampler` cannot be reused, for example: 
- https://github.com/vllm-project/vllm/pull/19482
- https://github.com/vllm-project/vllm/pull/26060
- https://github.com/vllm-project/vllm/pull/29223

#### Proposed Change:
- Delete `AscendRejctionSampler` and use `RejctionSampler` directly in model runner.
- Patch `RejctionSampler.expand_batch_to_tokens` and `RejctionSampler.rejection_sample`, maybe a better way is to make them as custom ops.
- Modify `NPUModelRunner` following https://github.com/vllm-project/vllm/pull/26060

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- [x] test logits processor for spec decoding
- [x] test logprobs for spec decoding
- [x] test logprobs for spec decoding + async shcheduling (test with https://github.com/vllm-project/vllm-ascend/pull/4893/)

- vLLM version: v0.12.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -1901,13 +1785,164 @@ def propose_draft_token_ids(sampled_token_ids):
...
+    ) -> tuple[
+            LogprobsLists | None,
+            list[list[int]],
+            dict[str, LogprobsTensors | None],
+            list[str],
+            dict[str, int],
+            list[int],
+    ]:
+        num_reqs = self.input_batch.num_reqs
+        discard_sampled_tokens_req_indices = np.nonzero(
```
Comment (zhenwenqi2024) [inline vllm_ascend/worker/model_runner_v1.py]: can some attrs like discard_requests_mask be deleted in prepare_inputs?

Comment (realliujiaxu) [reply vllm_ascend/worker/model_runner_v1.py]: we need to implement https://github.com/vllm-project/vllm/pull/28597 first.

File: vllm_ascend/patch/worker/__init__.py [hunk 2]
```diff
@@ -33,3 +33,4 @@
 import vllm_ascend.patch.worker.patch_qwen3_vl  # noqa
```
Comment (wangxiyuan) [inline vllm_ascend/patch/worker/__init__.py:33]: update this file please https://github.com/vllm-project/vllm-ascend/blob/main/vllm_ascend/patch/__init__.py
